components/ViewHabits.tsx

```diff
@@ -35,35 +53,113 @@ const styles = {
     maxHeight: '16vh',
     maxWidth: '100vw',
   }),
+
   habitsContainer: css({
-    maxWidth: '500px',
+    maxWidth: '700px',
     margin: '0 auto',
   }),
-  addHabitButton: css({
-    borderRadius: '7px',
-    position: 'absolute',
-    padding: '0',
-    height: '48px',
-    width: '48px',
-    top: '65px',
-    right: '8px',
-    transform: 'translateY(-50%)',
+  weeklyContainer: css({
+    display: 'flex',
+    flexDirection: 'column',
+    border: '2.5px solid #fff',
+
+    tr: {
+      display: 'flex',
+      flexDirection: 'row',
+
+      justifySelf: 'space-between',
+      alignSelf: 'space-between',
+    },
+
+    tbody: {
+      td: {
+        display: 'flex',
+        justifyContent: 'center',
+        alignItems: 'center',
+        flex: '1',
+        border: '0.5px solid #222',
+        padding: '0px 2px',
+        minHeight: '50px',
+        minWidth: '50px',
+        verticalAlign: 'middle',
+      },
+    },
+
+    th: {
+      display: 'flex',
+      justifyContent: 'center',
+      alignItems: 'center',
+      flex: '1',
+      border: '0.5px solid #222',
+      backgroundColor: '#0066ff',
+      padding: '0px 2px',
+      minHeight: '50px',
+      minWidth: '50px',
+      verticalAlign: 'middle',
+    },
+
+    '& :first-of-type': {
+      flexBasis: '50%',
+    },
   }),
 };
 
+type TimeView = 'daily' | 'weekly';
+
 const ViewHabits = () => {
   const { t } = useTranslation(['common', 'add-habit']);
   const [habitsData, setHabitsData] = useState<GoalWithHabitHistory[]>([]);
+  const [timeView, setTimeView] = useState<TimeView>('daily');
+  const [dates, setDates] = useState<Date[]>([]);
+
+  const daysToShow = 7;
+  const dateStrings = dates.map((date) => date.toDateString());
 
+  // Now fetch the data based on the number of days to show
+  // We don't want to fetch all habits data, as it would be inefficient to pull in potentially years' worth of data
+  // But this also means when saving data, these 7 days and any changes will have to be merged into the full set of entries
   useEffect(() => {
+    const date = new Date();
+    date.setDate(date.getDate() - (daysToShow - 1));
+
     habitsApi
-      .getHabits()
+      .getHabitsFromDate(date)
       .then(setHabitsData)
       .catch((error) => console.error(error));
   }, []);
 
+  // When the habits data changes, update the view
+  useEffect(() => {
+    const today = new Date();
+    const days = [];
+
+    // Get the last 7 days
+    for (let i = 0; i < daysToShow; i += 1) {
+      const newDay = new Date(today);
+      newDay.setDate(newDay.getDate() - i);
+
+      days.push(newDay);
+    }
```

Comment:
I think there might be a clearer way to do this - it took me a while to fully understand the intent. Perhaps it needs to be extracted into a named function outside/above this component, so that the reader does not need to focus on the specifics at this point

Reply:
Done this.